src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
+        switch (blockTag.getType()) {
+          case PARAM:
+            ParamTag paramTag = createParamTag(blockTag, parameters);
+            if (paramTag != null) {
+              // createParamTag returns null in case of generic type.
+              paramTags.add(paramTag);
+            }
+            break;
+          case RETURN:
+            returnTag = createReturnTag(blockTag);
+            break;
+          case EXCEPTION:
+          case THROWS:
+            throwsTags.add(createThrowsTag(classesInPackage, blockTag, sourceMember, className));
+            break;
+          default:
+            // ignore other block tags
+            break;
+        }
       }
+    }
+    return new BlockTags(paramTags, returnTag, throwsTags);
+  }
 
-      methods.add(
-          new DocumentedMethod(
-              new org.toradocu.extractor.Type(
-                  containingClass.containingPackage() + "." + containingClassName),
-              member.name(),
-              getReturnType(member),
-              getParameters(member),
-              memberParamTags,
-              member.isVarArgs(),
-              memberThrowsTags,
-              finalReturnTag));
+  /**
+   * Instantiate a tag of throws kind.
+   *
+   * @param classesInPackage list of class names in sourceMember's package
+   * @param blockTag the @throws or @exception Javadoc block comment containing the tag
+   * @param sourceMember the source member the tag refers to
+   * @param className qualified name of the class defining {@code sourceMember}
+   * @return the instantiated tag
+   * @throws ClassNotFoundException if the class of the exception type couldn't be found
+   */
+  private ThrowsTag createThrowsTag(
+      List<String> classesInPackage,
+      JavadocBlockTag blockTag,
+      CallableDeclaration<?> sourceMember,
+      String className)
+      throws ClassNotFoundException {
+    // Javaparser library does not provide a nice parsing of @throws tags. We have to parse the
+    // comment text by ourselves.
+    final Type blockTagType = blockTag.getType();
+    if (!blockTagType.equals(Type.THROWS) && !blockTagType.equals(Type.EXCEPTION)) {
+      throw new IllegalArgumentException(
+          "The block tag " + blockTag + " does not refer to an" + " @throws or @exception tag");
     }
 
-    return methods;
+    String comment = blockTag.getContent().toText();
+    final String[] tokens = comment.split("[\\s\\t]+", 2);
+    final String exceptionName = tokens[0];
+    Class<?> exceptionType =
+        findExceptionType(classesInPackage, sourceMember, exceptionName, className);
+    Comment commentObject = new Comment(tokens[1]);
+    return new ThrowsTag(exceptionType, commentObject);
   }
 
   /**
-   * Returns all non-private constructors and methods from the given {@code ClassDoc}. This includes
-   * inherited methods, except methods inherited from the class {@code java.lang.Object}.
+   * Instantiate a tag of return kind.
```

Comment:
Here, the Javadoc uses "instantiate" but the method name is "get".  That's inconsistent, and also inconsistent with the terminology used above.